sematic/resolvers/resource_managers/server_manager.py

```diff
@@ -1,15 +1,25 @@
 # Standard Library
-from typing import List
+import logging
+import time
+from threading import Thread
+from typing import List, Set
 
 # Sematic
 from sematic import api_client
 from sematic.plugins.abstract_external_resource import AbstractExternalResource
 from sematic.resolvers.abstract_resource_manager import AbstractResourceManager
 
+logger = logging.getLogger(__name__)
+
 
 class ServerResourceManager(AbstractResourceManager):
     """ResourceManager which uses server APIs to manage external resource metadata."""
 
+    def __init__(self, update_poll_interval_seconds: int = 600) -> None:
```

Comment:
Please extract to a constant.

10 minutes seems kinda much. Are these operations really that expensive? 1 minute, or even lower, would offer much better granularity and responsiveness for pipelines.